lib/src/components/NxDrawer/__tests__/NxDrawer.test.tsx

```diff
@@ -0,0 +1,270 @@
+    expect(dialog).toHaveAttribute('aria-modal', 'false');
+  });
+
+  it('renders children nodes within dialog', function() {
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const drawer = getDrawer({ children: <div data-testid="foo"/> })!;
+    expect(within(drawer).getByTestId('foo')).toBeInTheDocument();
+  });
+
+  it('merges any passed in className to the nx-drawer dialog', function() {
+    const drawerWithAddedClassName = getDrawer({ className: 'foo' });
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+    const drawer = getDrawer()!;
+
+    expect(drawerWithAddedClassName).toHaveClass('foo');
+
+    for (const cls of Array.from(drawer.classList)) {
+      expect(drawerWithAddedClassName).toHaveClass(cls);
+    }
+  });
+
+  it('includes any passed in attributes to the dialog element', function() {
+    const drawer = getDrawer({ id: 'drawer-id', lang: 'en_US' });
+    expect(drawer).toHaveAttribute('id', 'drawer-id');
+    expect(drawer).toHaveAttribute('lang', 'en_US');
+  });
+
+  it('should immediately set the dialog open attr when the open prop is set to true initially ' +
+  'and the open attr removed immediately when the open prop is set to false', async function() {
+    const { rerender } = quickRender({ open: true });
+
+    const dialog = screen.getByRole('dialog', { hidden: true });
+    await fireEvent.animationEnd(dialog);
+
+    expect(dialog).toHaveAttribute('open');
+
+    rerender(<NxDrawer open={false} onClose={() => {}} />);
+
+    expect(dialog).not.toHaveAttribute('open');
+  });
+
+  it('should set the dialog open attr only after the animation ends when the drawer is transitioning from ' +
+  'closed to open', async function() {
+    const { rerender } = quickRender({ open: false });
+
+    const dialog = screen.getByRole('dialog', { hidden: true });
+    expect(dialog).not.toHaveAttribute('open');
+
+    rerender(<NxDrawer open={true} onClose={() => {}} />);
+
+    expect(dialog).not.toHaveAttribute('open');
+
+    await fireEvent.animationEnd(dialog);
+
+    expect(dialog).toHaveAttribute('open');
+  });
+
+  describe('NxDrawer event listener support', () => {
+    it('executes onClose callback when ESC key is pressed', async function() {
+      const user = userEvent.setup();
+      const mockOnClose = jest.fn();
+
+      quickRender({ onClose: mockOnClose });
+
+      const dialog = screen.getByRole('dialog', { hidden: true });
+      await fireEvent.animationEnd(dialog);
+
+      expect(mockOnClose).not.toHaveBeenCalled();
+
+      await user.keyboard('{Escape}');
+
+      expect(mockOnClose).toHaveBeenCalled();
+    });
+
+    it('does not execute onClose if a key other than ESC is pressed', async function () {
+      const user = userEvent.setup();
+
+      const mockOnClose = jest.fn();
+
+      // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+      const drawer = getDrawer({ onClose: mockOnClose })!;
```

Comment:
To make the test realistic, should fire the animation so it's actually open during the typing

Reply:
> To make the test realistic, should fire the animation so it's actually open during the typing

Good point, I've updated it here: 3aa723913915f0855c0b3c04efddefb144b7260c